common/eth2/src/types.rs

```diff
@@ -244,6 +244,15 @@ pub struct ExecutionOptimisticForkVersionedResponse<T> {
     pub data: T,
 }
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct ExecutionOptimisticFinalizedForkVersionedResponse<T> {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub version: Option<ForkName>,
+    pub execution_optimistic: Option<bool>,
+    pub finalized: Option<bool>,
+    pub data: T,
+}
```

Comment:
I think the extra type is OK, because it makes the flags explicit. In future we might be able to consolidate all the `*ForkVersionedResponse` types, although it might come at the cost of some type safety/clarity. I.e. we could use `ExecutionOptimisticFinalizedForkVersionedResponse` for everything and we could set the `finalized` and `execution_optimistic` fields to `Some` where they are required (maybe using a builder pattern like you suggested).